ports/nrf/common-hal/_bleio/PacketBuffer.c

```diff
@@ -247,45 +250,62 @@ void common_hal_bleio_packet_buffer_construct(
     }
 }
 
-int common_hal_bleio_packet_buffer_readinto(bleio_packet_buffer_obj_t *self, uint8_t *data, size_t len) {
-    if (ringbuf_count(&self->ringbuf) < 2) {
+mp_int_t common_hal_bleio_packet_buffer_readinto(bleio_packet_buffer_obj_t *self, uint8_t *data, size_t len) {
+    if (ringbuf_num_filled(&self->ringbuf) < 2) {
         return 0;
     }
 
-    uint16_t packet_length;
-    ringbuf_get_n(&self->ringbuf, (uint8_t*) &packet_length, sizeof(uint16_t));
-
     // Copy received data. Lock out write interrupt handler while copying.
     uint8_t is_nested_critical_region;
     sd_nvic_critical_region_enter(&is_nested_critical_region);
 
+    // Get packet length, which is in first two bytes of packet.
+    uint16_t packet_length;
+    ringbuf_get_n(&self->ringbuf, (uint8_t*) &packet_length, sizeof(uint16_t));
```

Comment:
There is a bug here that needs an issue. If one catches the raised error and then calls readinto a second time the original packet length will be lost. This is an existing bug so I think it's ok to not fix here.

Reply:
This is easy to fix. I will just discard the entire too-big incoming packet.